### PR TITLE
[Console-IoT] Add ID to IoT Device Detail components

### PR DESCRIPTION
**Pull request by suyash-naithani**
### Type of change

- Refactoring

### Description

Add ID to IoT Device Detail components

### Checklist

- [ ] Update/write design documentation in `./documentation/design`
- [ ] Write tests and make sure they pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

File: console/console-init/ui/src/modules/iot-device-detail/components/ConfigurationInfo/ConfigurationInfo.tsx [hunk 1]
```diff
@@ -149,8 +150,9 @@ export const ConfigurationInfo: React.FC<IConfigurationInfoProps> = ({
                   <Card>
                     <CardBody>
                       <DropdownWithToggle
-                        id="ci-credential-type-dropdown"
-                        toggleId={"ci-credential-type-dropdown"}
+                        id="config-info-credential-type-dropdowntoggle"
```
Comment (rkpattnaik780) [inline console/console-init/ui/src/modules/iot-device-detail/components/ConfigurationInfo/ConfigurationInfo.tsx:153]: The id and toggleIDs for `DropdownWithToggle` component are vaguely named, any reason for that?

Comment (drcoolsanjeev) [reply console/console-init/ui/src/modules/iot-device-detail/components/ConfigurationInfo/ConfigurationInfo.tsx:153]: @ajaypratap003 Could you please suggest a good name for id and toggleId in this case?

Comment (obabec) [reply console/console-init/ui/src/modules/iot-device-detail/components/ConfigurationInfo/ConfigurationInfo.tsx:153]: I would add some information about where are we currently so probably 'device-config-info-credential-type-dropdowntoggle'. Some better idea @ajaypratap003 ?

File: console/console-init/ui/src/modules/iot-device-detail/components/ConfigurationInfo/ConfigurationInfo.tsx [hunk 2]
```diff
@@ -165,8 +167,9 @@ export const ConfigurationInfo: React.FC<IConfigurationInfoProps> = ({
                       />
                       {filterOptions?.length > 0 && (
                         <DropdownWithToggle
-                          id="ci-filter-dropdown"
-                          toggleId={"ci-filter-dropdown"}
+                          id="config-info-filter-dropdowntoggle"
```
Comment (obabec) [inline console/console-init/ui/src/modules/iot-device-detail/components/ConfigurationInfo/ConfigurationInfo.tsx:170]: Same as previous comment.

File: console/console-init/ui/src/modules/iot-device-detail/components/ConfigurationInfo/ConfigurationInfo.tsx [hunk 3]
```diff
@@ -180,7 +183,7 @@ export const ConfigurationInfo: React.FC<IConfigurationInfoProps> = ({
                 </GridItem>
               </Grid>
               <CredentialsView
-                id="ci-credentials-view"
+                id="config-info-credentials-view"
```
Comment (obabec) [inline console/console-init/ui/src/modules/iot-device-detail/components/ConfigurationInfo/ConfigurationInfo.tsx:186]: Same as previous comment.

File: console/console-init/ui/src/modules/iot-device-detail/components/CredentialsView/ExtensionsView/ExtensionsView.tsx [hunk 4]
```diff
@@ -82,7 +82,7 @@ export const ExtensionsView: React.FC<IExtensionsViewProps> = ({
             </Title>
           </GridItem>
           <Table aria-label="extensions view table" cells={columns} rows={rows}>
-            <TableHeader />
+            <TableHeader id="extensions-view-table" />
```
Comment (obabec) [inline console/console-init/ui/src/modules/iot-device-detail/components/CredentialsView/ExtensionsView/ExtensionsView.tsx:85]: Following pattern you have chosen it should be 'extensions-view-table-header / extensions-view-tableheader'

Comment (ajaypratap003) [reply console/console-init/ui/src/modules/iot-device-detail/components/CredentialsView/ExtensionsView/ExtensionsView.tsx:85]: Good to go with `id="extensions-view-tableheader"`

File: console/console-init/ui/src/modules/iot-device-detail/components/ConfigurationInfo/ConfigurationInfo.tsx (hunk 3 above)
Comment (ajaypratap003) [reply console/console-init/ui/src/modules/iot-device-detail/components/ConfigurationInfo/ConfigurationInfo.tsx:186]: I would suggest id here
` id='device-detail-config-info-credentials-view'` 
@obabec wdyt?

File: console/console-init/ui/src/modules/iot-device-detail/components/ConfigurationInfo/ConfigurationInfo.tsx (hunk 2 above)
Comment (ajaypratap003) [reply console/console-init/ui/src/modules/iot-device-detail/components/ConfigurationInfo/ConfigurationInfo.tsx:170]: I would suggest id here
` id='device-detail-config-info-credential-filter-dropdown'` 
` toggleId='device-detail-config-info-credential-fillter-dropdowntoggle'` 
@obabec wdyt?

File: console/console-init/ui/src/modules/iot-device-detail/components/ConfigurationInfo/ConfigurationInfo.tsx (hunk 1 above)
Comment (ajaypratap003) [reply console/console-init/ui/src/modules/iot-device-detail/components/ConfigurationInfo/ConfigurationInfo.tsx:153]: @obabec This page belongs to device details page in which we have a  configuration info tab and DropdownToggle is used for credentials filter. First dopdown have some options like all credentials, enabled, password etc. and scond one will have the options based on first dropdown selection.
I would suggest id here
` id='device-detail-config-info-credential-type-dropdown'` 
` toggleId='device-detail-config-info-credential-type-dropdowntoggle'` 
@obabec wdyt?

image

File: console/console-init/ui/src/modules/iot-device-detail/DeviceDetailPage.tsx [hunk 5]
```diff
@@ -95,7 +95,7 @@ export default function DeviceDetailPage() {
     () => (
       <Breadcrumb>
         <BreadcrumbItem>
-          <Link id="cdetail-link-home" to={"/"}>
+          <Link id="device-detail-home-link" to={"/"}>
```
Comment (drcoolsanjeev) [inline console/console-init/ui/src/modules/iot-device-detail/DeviceDetailPage.tsx:98]: @ajaypratap003 for being consistent in breadcrumb i suggest to keep the id same at all places in the whole app. WDYT ?

File: console/console-init/ui/src/modules/iot-device-detail/components/DeviceDetailHeader/DeviceDetailHeader.tsx [hunk 6]
```diff
@@ -221,7 +221,7 @@ const DeviceDetailHeader: React.FunctionComponent<IDeviceDetailHeaderProps> = ({
         id="device-detail-header-kebab-dropdown"
         isPlain={true}
         position={DropdownPosition.right}
-        toggleId="device-detail-header-kebab-toggle"
+        toggleId="device-detail-header-dropdown-toggle"
```
Comment (drcoolsanjeev) [inline console/console-init/ui/src/modules/iot-device-detail/components/DeviceDetailHeader/DeviceDetailHeader.tsx:224]: to be consistent at all places revisit as id=`device-detail-header-dropdowntoggle`

File: console/console-init/ui/src/modules/iot-device-detail/DeviceDetailPage.tsx (hunk 5 above)
Comment (suyash-naithani) [reply console/console-init/ui/src/modules/iot-device-detail/DeviceDetailPage.tsx:98]: @drcoolsanjeev What ID should we keep for all of them?

**Pull request merged**